### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by fortunatus-png**
Add tests for core functionality.

### CODE REVIEW

**Comment by fortunatus-png** [comment]
The changes are ready. I'd appreciate your review.

File: tests/example.spec.js [hunk 1]
```diff
@@ -3,20 +3,45 @@ import { test, expect } from '@playwright/test';
 test('has title', async ({ page }) => {
   await page.goto('/');
   await expect(page).toHaveTitle("To-Do's");
+  await expect(page.getByRole('heading', { name: 'To-Do\'s 📜' })).toBeVisible();
 });
 
 test('add task', async ({ page }) => {
```
Comment (madebybee) [inline tests/example.spec.js]: Typically you would wrapp tests that belong to the same thing in a describe block.

File: tests/example.spec.js [hunk 2]
```diff
@@ -3,20 +3,45 @@ import { test, expect } from '@playwright/test';
...
   await page.goto('/');
```
Comment (madebybee) [inline tests/example.spec.js]: You are repeating goto in every test. You can use beforeEach for that.

File: tests/example.spec.js [hunk 3]
```diff
@@ -3,20 +3,45 @@ import { test, expect } from '@playwright/test';
...
+  await page.getByRole('textbox').fill('aaa');
```
Comment (madebybee) [inline tests/example.spec.js]: You could put aaa into a variable so that you can later check on it and not repeat the string.

File: tests/example.spec.js [hunk 4]
```diff
@@ -3,20 +3,45 @@ import { test, expect } from '@playwright/test';
...
 
-test('input field', async ({ page }) => {
+test('clear all tasks', async ({ page }) => {
   await page.goto('/');
-  await page.getByRole('textbox').fill('Add your task');
+  await page.getByRole('textbox').fill('aaa');
+  await page.getByRole('button', { name: '✚' }).click();
+  await page.getByRole('button', { name: 'Clear All' }).click();
+  await expect(page.getByText('aaa')).not.toBeVisible();
 });
```
Comment (madebybee) [inline tests/example.spec.js:77]: Congrats on writing your first e2e test! Looks good so far and it does what it should :)

The next pro move would be to make a POM (Page Object Model).
You can read about it in the documentation.

The essential goal of it is that each page of your application has an object with elements and functions that you can reuse. For example properties for addButton, clearButton and functions like addTask or deleteTask.

If you define these, your tests will much more be readable like a story.

Comment (madebybee) [reply tests/example.spec.js:77]: Another thing is that localStorage is not tested yet.

Comment (madebybee) [reply tests/example.spec.js:77]: Edge Cases like when user types a space or nothing and tries to add a task are not covered. In a professional setting exactly these are very important since user do weird things with apps :)

**Review by madebybee — CHANGES_REQUESTED** [review]

File: tests/example.spec.js (hunk 4 above)
Comment (fortunatus-png) [reply tests/example.spec.js:77]: I’ve applied your suggestion. Could you please review it?

**Review by madebybee — APPROVED** [review]
looks goood!
